noxfile.py

```diff
@@ -64,31 +65,31 @@ def stable_version_without_post_migration(session: nox.Session):
     pre_migration = get_pre_migration(latest_tag)
 
     with tempfile.TemporaryDirectory() as temp_dir:
-        requirements_file = pathlib.Path(temp_dir) / "requirements.txt"
+        temp_path = pathlib.Path(temp_dir)
+        base_path = pathlib.Path(os.path.abspath("."))
+        shutil.copytree(base_path / "tests", temp_path / "tests")
+        shutil.copy(base_path / "pyproject.toml", temp_path / "pyproject.toml")
+        shutil.copy(base_path / "poetry.lock", temp_path / "poetry.lock")
+        os.chdir(temp_dir)
```

Comment:
https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.MonkeyPatch.chdir so it will be revrted at the end of the test